server/middleware.go

```diff
@@ -8,13 +8,24 @@ import (
 
 // Location middleware
 func WhereaboutsRequired() gin.HandlerFunc {
```

Comment:
I changed this around to look for a `query` parameter on the request (I had planned to do this a little while ago #117).

The issue explains in more detail, but the tl;dr is it saves us making an extra database call and means we don't need to flush cache each time the user changes their settings.